FirstLegoLeague/fllscoring
# Common ranking

Part of #245: Move Ranking computation to common code (used by ng-scores or a new ng-ranking service)

All tests pass for each commit.

The first few commits are only refactors, no functional changes.
The last 2 fix two small bugs.

To test:
- For the first commits: check that the rankings still work in the frontend
- For the negative round bug: edit a score to have a round of 0 or lower, verify that it no longer ends up in rankings
- For the crash in CSV export bug: edit a score to have a non-existing team number, verify that there are no errors in JS Console, and Export button on Ranking screen works (team info will be empty for that score, though)
